.github/workflows/docker-build-push.yml

```diff
@@ -26,11 +26,11 @@ jobs:
 
       - name: Checkout pages branch and sync with changes
         run: |
-          echo ${BRANCH_NAME}
-          if [${BRANCH_NAME} == 'refs/heads/master']
+          echo ${GITHUB_REF#refs/*/}
+          if [${GITHUB_REF#refs/*/} == 'refs/heads/master']
           then
             pages_branch="gh-pages"
-          elif [${BRANCH_NAME} == 'refs/heads/staging']
+          elif [${GITHUB_REF#refs/*/} == 'refs/heads/staging']
           then
             pages_branch="gh-pages-stages"
           fi
```

Comment:
Can't make this a suggestion, but try going back to: 
```
 run: |
          echo $GITHUB_REF
          if [ $GITHUB_REF == 'refs/heads/master' ]
          then
            pages_branch="gh-pages"
          elif [ $GITHUB_REF == 'refs/heads/staging' ]
          then
            pages_branch="gh-pages-stages"
          fi
```
And change line back 92 to 
```
if: github.ref == 'refs/heads/staging'
```
and similarly line 100
```
if: github.ref == 'refs/heads/master'
```